avail.go

```diff
@@ -0,0 +1,254 @@
+	config Config
+	ctx    context.Context
+}
+
+// NewAvailDA returns an instance of AvailDA
+func NewAvailDA(config Config, ctx context.Context) *AvailDA {
+	return &AvailDA{
+		ctx:    ctx,
+		config: Config{LcURL: config.LcURL, AppID: config.AppID},
+	}
+}
+
+var _ da.DA = &AvailDA{}
+
+// MaxBlobSize returns the max blob size
+func (c *AvailDA) MaxBlobSize() (uint64, error) {
+	var maxBlobSize uint64 = 64 * 64 * 500
+	return maxBlobSize, nil
+}
+
+// Submit each blob to avail data availability layer
+func (c *AvailDA) Submit(daBlobs []da.Blob) ([]da.ID, []da.Proof, error) {
+	resultChan := make(chan SubmitResponse, len(daBlobs))
+	errorChan := make(chan error, len(daBlobs))
+
+	var wg sync.WaitGroup
+
+	var mu sync.Mutex
+
+	for _, blob := range daBlobs {
+		wg.Add(1)
+
+		// Start a goroutine for each blob
+		go func(blob da.Blob) {
+			defer wg.Done()
+			encodedBlob := base64.StdEncoding.EncodeToString(blob)
+			requestData := SubmitRequest{
+				Data: encodedBlob,
+			}
+
+			requestBody, err := json.Marshal(requestData)
+			if err != nil {
+				errorChan <- err
+				return
+			}
+
+			// Make a POST request to the /v2/submit endpoint.
+			response, err := http.Post(c.config.LcURL+"/submit", "application/json", bytes.NewBuffer(requestBody))
+			if err != nil {
+				errorChan <- err
+				return
+			}
+
+			defer func() {
+				err = response.Body.Close()
+				if err != nil {
+					log.Println("error closing response body", err)
+				}
+			}()
+
+			responseData, err := io.ReadAll(response.Body)
+			if err != nil {
+				errorChan <- err
+				return
+			}
+
+			var submitResponse SubmitResponse
+			err = json.Unmarshal(responseData, &submitResponse)
+			if err != nil {
+				errorChan <- err
+				return
+			}
+
+			// Acquire the mutex before updating slices
+			mu.Lock()
+			resultChan <- SubmitResponse{
+				BlockNumber:      submitResponse.BlockNumber,
+				BlockHash:        submitResponse.BlockHash,
+				TransactionHash:  submitResponse.TransactionHash,
+				TransactionIndex: submitResponse.TransactionIndex,
+			}
+			mu.Unlock()
+
+		}(blob)
+	}
+
+	go func() {
+		wg.Wait()
+		close(resultChan)
+		close(errorChan)
+	}()
+
+	// Collect results from channels
+	var ids []da.ID
+	var proofs []da.Proof
+
+	for result := range resultChan {
+		ids = append(ids, makeID(result.BlockNumber))
+		proofs = append(proofs, makeProofs(result.TransactionHash))
+	}
+
+	// Check for errors
+	if err := <-errorChan; err != nil {
+		return nil, nil, err
+	}
+
+	fmt.Println("successfully submitted blobs to avail")
+	return ids, proofs, nil
+}
+
+// Get returns Blob for each given ID, or an error
+func (c *AvailDA) Get(ids []da.ID) ([]da.Blob, error) {
+	var blobs [][]byte
+	var blockNumber uint32
+	for _, id := range ids {
+	Loop:
+		blockNumber = binary.BigEndian.Uint32(id)
+		blocksURL := fmt.Sprintf(c.config.LcURL+BlockURL, blockNumber)
+		parsedURL, err := url.Parse(blocksURL)
+		if err != nil {
+			return nil, err
+		}
+		time.Sleep(10 * time.Second)
```

Comment:
This sleeps in the first iteration as well, right? Might be better to move this to the `else ... == PROCESSING_BLOCK` condition.